modules/basics/src/main/java/com/opengamma/basics/date/PeriodAdjustment.java

```diff
@@ -162,7 +162,7 @@ private void validate() {
   public LocalDate adjust(LocalDate date) {
     ArgChecker.notNull(date, "date");
     LocalDate unadjusted = additionConvention.adjust(date, period, adjustment.getCalendar());
-    return unadjusted.with(adjustment);
+    return adjustment.adjust(unadjusted);
```

Comment:
New way round calls a specific method typed to accept LocalDate, whereas the previous way had to check the object multiple times and call logic via an interface. Faster.